src/utils/GitHubUtils/GitHubUtils.js

```diff
@@ -61,56 +61,58 @@ export const setEmptyCalendarValues = calendar => {
   return copiedCalendar;
 };
 
-export const mergeCalendarsContributions = (
-  actualCalendar,
-  gitHubUserJsonCalendar,
-  timeframe
-) => {
+export const sanitize = calendar =>
+  calendar.children[0].children
+    .filter(item => {
+      const isDay =
+        item.attributes.class !== "wday" && item.attributes.class !== "month";
```

Comment:
Isn't there a specific class for days that you can check for? This way if they introduce a new class this would break.

Also why isn't `wday` a day? :)

Reply:
Let's first understand the structure then.

![Screenshot 2020-04-16 at 10 46 50](https://user-images.githubusercontent.com/30864202/79435163-9bf98400-7fcf-11ea-8566-9512b19b7d1d.png)

The "item" (from `filter(item => ...`) refers to either a "week" or a text around the calendar (text == months on the top and days on the left side). 

![Screenshot 2020-04-16 at 10 50 56](https://user-images.githubusercontent.com/30864202/79435593-2a6e0580-7fd0-11ea-9921-091e659a9e9b.png)

Each "week" has a `children` prop which represents the days of "that" week. For example, the 52nd element refers to the last week, i.e. the last column within the calendar; this week has 5 children element, that's why the last column consists of 5 squares (as opposed to the column to it which have 7 children consistently).

Starting from the 53rd item, they no longer refer to "weeks"; those items are the months and days around the calendar even though they have a `children` prop as well (but this prop always consists of a single "text" element). What makes them distinguishable from weeks is the "class" attribute they have (`month` for months and `wday` for the days on the left side).

As for the questions:

> Isn't there a specific class for days that you can check for? This way if they introduce a new class this would break.

We may use the `name` attribute because that's also different (it's `g` in the case of weeks whereas it's `element` in the case of the texts). But I'd say using the `attributes.class` to differentiate between "genuine" weeks and texts is the ideal approach.

> Also why isn't wday a day? :)

The clarification above should answer this question.

Reply:
Thanks, it's clear now. I'd still propose to do a positive check instead of a negative one. So either check for the name or perhaps select the first 52 elements (if that's a constant pattern). Regarding whether this is needed or not, I think this is just a filter that might not be needed at all, for example, if you iterate on weeks based on indexes you don't need to care about indexes higher than 53. This is what I meant here: https://github.com/c-hive/team-contribution-calendar/pull/70#discussion_r408412767

Reply:
If this is needed, I'd use `filter` that is more expressive in this case imo.

Reply:
> If this is needed, I'd use filter that is more expressive in this case imo.

We're already using `filter()` but it's combined with `map()` to convert the object of arrays to an array of arrays structure. Perhaps the conversion isn't necessary as removing the "noise" is a fairly nice improvement per se.

> So either check for the name or perhaps select the first 52 elements (if that's a constant pattern).

It appears to be a constant pattern. Then theoretically we can use `slice()` to cut off the elements that are supposed to be disregarded (i.e. the months and days).

Reply:
> This results in the exact same structure as currently except if the array's length varies.

On second thoughts, it's not the exact same structure but I'll look into it.

Reply:
> We're already using filter() but it's combined with map() to convert the array of objects to an array of arrays structure.

Either way you could just add another fliter.

Reply:
https://github.com/c-hive/team-contribution-calendar/pull/70#issuecomment-615214224